File: KenKenAndroid/src/com/anthonysottile/kenken/SignNumber.java
package com.anthonysottile.kenken;

import org.json.JSONException;
import org.json.JSONObject;

public class SignNumber {

	private Sign sign;
	public Sign getSign() {
		return this.sign;
	}

	private int number;
	public int getNumber() {
		return this.number;
	}

	@Override
	public String toString() {
		StringBuilder sb = new StringBuilder();
		sb.append(this.number);
		sb.append(' ');
		sb.append(this.sign.toString());

		return sb.toString();    }

	public SignNumber(Sign sign, int number) {
		this.sign = sign;
		this.number = number;
	}

	// #region JSON Serialization

	private static final String signProperty = "Sign";
	private static final String numberProperty = "Number";

	public SignNumber(JSONObject json) {

		try {
			this.sign = Sign.toSign(json.getInt(SignNumber.signProperty));
			this.number = json.getInt(SignNumber.numberProperty);
		} catch (JSONException e) {
			e.printStackTrace();
		}
	}

	public JSONObject ToJson() {
		JSONObject json = new JSONObject();

		try {
			json.put(SignNumber.signProperty, this.sign.getIntValue());
			json.put(SignNumber.numberProperty, this.number);
		} catch (JSONException e) {
			e.printStackTrace();
		}

		return json;
	}

	// #endregion
}

File: app/src/main/java/com/anthonysottile/kenken/SignNumber.kt
package com.anthonysottile.kenken

import org.json.JSONObject

class SignNumber {
    val sign: Sign
    val number: Int

    override fun toString(): String {
        return "${this.number} ${this.sign}"
    }

    constructor(sign: Sign, number: Int) {
        this.sign = sign
        this.number = number
    }

    constructor(json: JSONObject) {
        this.sign = Sign.toSign(json.getInt(SignNumber.signProperty))
        this.number = json.getInt(SignNumber.numberProperty)
    }

    fun toJson(): JSONObject {
        val json = JSONObject()
        json.put(SignNumber.signProperty, this.sign.intValue)
        json.put(SignNumber.numberProperty, this.number)
        return json
    }

    companion object {
        private const val signProperty = "Sign"
        private const val numberProperty = "Number"
    }
}
